Sources/Datadog/Tracing/Span/SpanEncoder.swift

```diff
@@ -99,6 +102,15 @@ internal struct SpanEncoder {
         case mobileNetworkCarrierAllowsVoIP = "meta.network.client.sim_carrier.allows_voip"
     }
 
+    /// Coding keys for dynamic `Span` attributes specified by user.
+    private struct DynamicCodingKey: CodingKey {
+        var stringValue: String
+        var intValue: Int?
+        init?(stringValue: String) { self.stringValue = stringValue }
+        init?(intValue: Int) { return nil }
```

Comment:
why do we need this line?

Reply:
This is required to conform to [`CodingKey` protocol](https://developer.apple.com/documentation/swift/codingkey). It's taken from `LogEncoder`, where I didn't found different way around 😞.